networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/Eth2Peer.java

```diff
@@ -171,9 +180,16 @@ void markChainValidated() {
   }
 
   public SafeFuture<UnsignedLong> sendPing() {
-    return requestSingleItem(rpcMethods.ping(), metadataMessagesFactory.createPingMessage())
-        .thenApply(PingMessage::getSeqNumber)
-        .thenPeek(this::updateMetadataSeqNumber);
+    if (outstandingPings.getAndIncrement() >= outstandingPingsThreshold) {
+      LOG.debug("Disconnecting the peer {} due to PING timeout.", getId());
+      disconnectImmediately();
```

Comment:
Even though the peer is probably gone, I don't think it would hurt to send a goodBye message just in case.